cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/RemoteDatasetFramework.java

```diff
@@ -199,7 +200,8 @@ public void deleteAllInstances(Id.Namespace namespaceId) throws DatasetManagemen
   public <T extends Dataset> T getDataset(
     Id.DatasetInstance datasetInstanceId, Map<String, String> arguments,
     @Nullable ClassLoader classLoader,
-    @Nullable Iterable<? extends Id> owners) throws DatasetManagementException, IOException {
+    @Nullable Iterable<? extends Id> owners)
+    throws DatasetManagementException, IOException, ServiceUnavailableException {
```

Comment:
``ServiceUnavailableException`` is unchecked. It just got removed in another PR. You may have conflicts.